File: CompanyRatingProject/DTO/ModelViewsObjects/CompanyDTO.cs
namespace DTO.ModelViewsObjects;

public class CompanyDto
{
    public string Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public double Grade1Avg { get; set; }
    public double Grade2Avg { get; set; }
    public double Grade3Avg { get; set; }
    public double Grade4Avg { get; set; }

    public double Grade5Avg { get; set; }
    public double TotalAvg { get; set; }
    public ICollection<RatingDto> Ratings { get; set; }
}

File: CompanyRatingProject/CompanyRatingProject/ClientApp/src/app/DTO/CompanyDTO.ts
import {RatingDTO} from "./RatingDTO";

export class CompanyDTO{
    constructor(
        public id?:string,
        public name?:string,
        public description?:string,
        public grade1Avg?:number,
        public grade2Avg?:number,
        public grade3Avg?:number,
        public grade4Avg?:number,
        public grade5Avg?:number,
        public totalAvg?:number,
        public ratings?:RatingDTO[]
    ) {}
}